pkg/template/event.go

```diff
@@ -33,16 +34,16 @@ func ResolveParams(rt ResolvedTrigger, body []byte, header http.Header) ([]trigg
 		return nil, fmt.Errorf("error merging trigger params: %w", err)
 	}
 
-	out, err = applyEventValuesToParams(out, body, header)
-	if err != nil {
-		return nil, fmt.Errorf("failed to ApplyEventValuesToParams: %w", err)
-	}
-
 	var ttParams []triggersv1.ParamSpec
 	if rt.TriggerTemplate != nil {
 		ttParams = rt.TriggerTemplate.Spec.Params
 	}
 
+	out, err = applyEventValuesToParams(out, body, header, ttParams)
+	if err != nil {
+		return nil, fmt.Errorf("failed to ApplyEventValuesToParams: %w", err)
+	}
+
 	return MergeInDefaultParams(out, ttParams), nil
```

Comment:
do we still need to call `MergeInDefaultParams` since we are now adding default values in `applyEventValuesToParams` itself?

Reply:
Yes we do.  Unit tests fail without it:

```
using default defaultVal since failed to replace JSONPath value for param p2: $(body.p2): error: p2 is not found
--- FAIL: TestResolveParams (0.00s)
    --- FAIL: TestResolveParams/add_default_values_for_params_with_missing_values (0.00s)
        event_test.go:351: didn't get expected params -want + got:   []v1alpha1.Param{
              	{Name: "p1", Value: "val1"},
            - 	{Name: "p2", Value: "defaultVal"},
              }
FAIL
FAIL	github.com/tektoncd/triggers/pkg/template	0.062s
FAIL
```

Reply:
Hmmm, interesting. I think that makes sense. But makes me wonder if we can simplify this code somehow.

Reply:
ok @dibyom I've pushed a separate commit that consolidates `applyEventValuesToParams` and `mergeInDefaultParams`

it is possible the changes pulled in with the rebase helped things (the method was minimally switched from public to package only)

otherwise, the biggest takeaway is that an explicit for loop on the `default` array has been replaced with a golang map lookup.... that is positive.

If you like this approach better I'll  squash the commits.